Utilities/File.cpp

```diff
@@ -2486,16 +2485,62 @@ bool fs::pending_file::commit(bool overwrite)
 		disp.DeleteFileW = false;
 		ensure(SetFileInformationByHandle(file.get_handle(), FileDispositionInfo, &disp, sizeof(disp)));
 	}
+
+	std::vector<std::wstring> hardlinkPaths;
+
+	const auto ws1 = to_wchar(m_path);
+
+	const HANDLE fileHandle = CreateFileW(ws1.get(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nullptr, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, nullptr);
+
+	while (fileHandle != INVALID_HANDLE_VALUE)
+	{
+		// Get file ID (used to check for hardlinks)
+		BY_HANDLE_FILE_INFORMATION fileInfo;
+
+		if (!GetFileInformationByHandle(fileHandle, &fileInfo) || fileInfo.nNumberOfLinks == 1)
+		{	
+			CloseHandle(fileHandle);
+			break;
+		}
+
+		// Buffer for holding link name
+		WCHAR linkNameBuffer[MAX_PATH];
+		HANDLE findHandle = FindFirstFileNameW(ws1.get(), 0, nullptr, linkNameBuffer);
```

Comment:
Two things:
-Limiting to MAX_PATH on windows has been a problem in the past if I recall, we should avoid that if possible
-Is nullptr for StringLength even valid, documentation doesn't mention it

Reply:
Fixed